satellite/metainfo/metainfo.go

```diff
@@ -506,3 +509,34 @@ func CreatePath(ctx context.Context, projectID uuid.UUID, segmentIndex int64, bu
 	}
 	return storj.JoinPaths(entries...), nil
 }
+
+// checks if bucket has any pointers(entries)
+func (endpoint *Endpoint) checkBucketPointers(ctx context.Context, req *pb.SegmentInfoRequest) (*pb.SegmentInfoResponse, error) {
+	// not_found error indicates connectory id set but no entry exists
+	resp, err := endpoint.SegmentInfo(ctx, req)
+	return resp, err
+}
+
+// ValueAttributeInfo commits segment metadata
+func (endpoint *Endpoint) ValueAttributeInfo(ctx context.Context, req *pb.SegmentInfoRequest) (*pb.SegmentInfoResponse, error) {
```

Comment:
I'm not sure why we need a separate call to metainfo. Why we cannot do this as a part of commiting segment?

also I'm not sure if its a good idea to reuse `pb.SegmentInfoRequest`

Reply:
Thats how I had the implementation done in commitsegment. I talked to Dennis about the DB hits, it is going to create for every segment. So after talking to Dennis, we agreed to make it as an endpoint and my understanding is on the client's side, the uplink logic should be able to handle it. 

Regarding putting in segmentInfo, we will be calling a segmentInfo function to know if any info about the bucket id exists and then save the value attribution, so either I create a separate value attribution message and call segment info or put value attribution info into segment info, it is just a matter of preference...let me know what you ur thoughts are. ... I can create a separate message called value attribution and have to call in segment info in it. let me know what your thoughts are?